components/channel_invite_modal/channel_invite_modal.tsx

```diff
@@ -229,6 +231,12 @@ export default class ChannelInviteModal extends React.PureComponent<Props, State
         return option.username;
     }
 
+    private filterOutDeletedAndExcludedAndNotInTeamUsers = (users: UserProfile[], excludeUsers: UserProfile[]): UserProfileValue[] => {
+        return users.filter((user) => {
+            return user.delete_at === 0 && !excludeUsers.map((user) => user.id).includes(user.id);
```

Comment:
Filtering out users that are in `excludeUsers` this way isn't great because, for every user in `users`, we're looping over `excludeUsers` twice: once to turn it into an array of IDs and a second time to see if it matches this that user's ID.

Instead, we could do this a bit more efficiently by having `excludeUsers` be a Set of user IDs to begin with since that would mean we could check if the user's ID is in it more quickly (since `Set.has` is faster than `Array.includes`) and we would only need to go over the array once to turn it into a Set. We're already doing some work in the `render` method to create `excludedAndNotInTeamUsers` from a couple of arrays, so we can just change that to instead make a Set out of `props.profilesNotInCurrentTeam` and add all the users in `props.excludeUsers` to it if that exists